VIMC-4734 Add comments to responsibilities

mwoodbri opened the pull request:
Summary:
* Fix build: has been failing since https://github.com/vimc/orderly/blame/master/inst/create_orderly_demo.sh#L8
* Fix test: [sorting](https://github.com/vimc/montagu-webapps/blob/master/app/src/main/contrib/components/Responsibilities/Overview/List/ResponsibilityList.tsx#L34) of scenarios without descriptions non-deterministic
* Add modal to allow comments to be created/reviewed
* Requires vimc/montagu-db#159 and vimc/montagu-api#458

To test:
* Spin up the admin app, visit http://localhost:5000/admin/touchstones/op-2018/op-2018-1/responsibilities/ and add some comments

To-do:
- [x] Further tests for modal
- [x] Integration test
- [x] Update config/submodules to point at `master`

EmmaLRussell commented on app/src/main/admin/actions/adminTouchstoneActionCreators.ts:
```diff
@@ -30,6 +39,36 @@ export const adminTouchstoneActionCreators = {
+            const service = new TouchstonesService(dispatch, getState);
+            const result = await service.addResponsibilityComment(touchstoneVersion, modellingGroupId, scenarioId, comment);
+            if (result) {
+                dispatch(this.setCurrentTouchstoneResponsibility(null));
```
I suppose there's an argument that setting current touchstone responsibility back to null could be initiated directly by the modal on save as it is in cancel, to decouple these actions from each other - but I don't mind too much!

EmmaLRussell reviewed (commented):
This looks great to me, works likes a charm. Haven't pored through all the tests as I know you're still working on those.

mwoodbri commented:
Changes since last commit:
* Update to match POST endpoint changes in `montagu-api`
* Decouple action dispatch
* Add integration tests
* Expand other tests

EmmaLRussell reviewed (approved):
All looking good. I guess we might end up putting a friendlier datetime format on the comment dialog, but I think this is fine to start with.

EmmaLRussell commented on app/src/main/shared/services/TouchstonesService.ts:
```diff
@@ -28,7 +29,11 @@ export class TouchstonesService extends AbstractLocalService {
 
     addResponsibilityComment(touchstoneVersion: string, modellingGroupId: string, scenarioId: string, comment: string) {
-        return this.post(`/touchstones/${touchstoneVersion}/responsibilities/comments/?group_id=${modellingGroupId}&scenario_id=${scenarioId}&comment=${encodeURIComponent(comment)}`)
+        const responsibilityComment: ResponsibilityComment = { comment, added_by: undefined, added_on: undefined };
```
Ah yeah, this is already outdated!

mwoodbri replied:
> Ah yeah, this is already outdated!

Yes - the use of `undefined` should really have been the signal for a re-think, so thanks. We've ended up in a better place.
